cmd/influx/write.go

```diff
@@ -184,10 +208,18 @@ func (b *writeFlagsBuilder) createLineReader(ctx context.Context, cmd *cobra.Com
 				return nil, csv2lp.MultiCloser(closers...), fmt.Errorf("failed to open %q: %v", file, err)
 			}
 			closers = append(closers, f)
-			readers = append(readers, decode(f), strings.NewReader("\n"))
+
+			compressed := b.Compression == "gzip" || (len(b.Compression) == 0 && strings.HasSuffix(file, ".gz"))
+			if compressed {
+				file = strings.TrimSuffix(file, ".gz")
```

Comment:
As a user I would prefer to see the `.gz` suffix in the error message. Not blocking.